portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +14,37 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
+import java.util.ArrayList;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.*;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+    ArrayList<String> Greetings = new ArrayList();
+    ArrayList<String> commentRepo = new ArrayList<String>();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+      
   }
+  @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse res) throws IOException{
+        Entity commentEntity = new Entity("comment");
+        String jsonstring = req.getParameter("comment");
+        commentEntity.setProperty("comment",jsonstring);
+        long timestamp = System.currentTimeMillis();
+        commentEntity.setProperty("timestamp", timestamp);
+        datastore.put(commentEntity);
+        commentRepo.add(jsonstring);
+        System.out.println(commentRepo);
+        res.sendRedirect("/index.html");
```

Comment:
Break up code into logical sections (newlines). Does datastore.put throw an exception? If so, can you put a try / catch around it.